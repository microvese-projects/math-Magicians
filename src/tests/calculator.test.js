import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

test('Should render calculator correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
