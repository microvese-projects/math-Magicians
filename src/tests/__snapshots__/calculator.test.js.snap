// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render calculator correctly 1`] = `
<section
  id="calculator"
>
  <div
    id="output"
  >
    <input
      readOnly={true}
      type="text"
      value={0}
    />
    <input
      readOnly={true}
      type="text"
      value={0}
    />
  </div>
  <div
    id="inputs-container"
  >
    <button
      className="btn normal"
      id="allClear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn normal"
      id="addSubtract"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn normal"
      id="remainder"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn colored"
      id="divide"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn colored"
      id="multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn colored"
      id="subtract"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn normal"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn colored"
      id="add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn normal dbl"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn normal"
      id="period"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn colored"
      id="equals"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
